R/filter.R

```diff
@@ -33,3 +34,22 @@ porcelain_filter_metadata <- function(req, res) {
   req$received_time <- now_utc()
   plumber::forward()
 }
+
+
+porcelain_filter_request_id <- function(logger) {
+  force(logger)
+  function(req, res) {
+    if (!is.null(req$HTTP_X_REQUEST_ID)) {
+      request_id <- req$HTTP_X_REQUEST_ID
+    } else {
+      request_id <- ids::uuid()
+    }
```

Comment:
completely up to you but i think this is cleaner
```suggestion
    request_id <- req$HTTP_X_REQUEST_ID %||% ids::uuid()
```